### PR TITLE
Add custom content types, future proof classification model

### PR DESCRIPTION
**Pull request by tonistuulNG**
Requesting feedback on the classification model, wont merge before consensus

### CODE REVIEW

**Review by joonasroosalung — APPROVED** [review]

File: docs/technical/api-spec.yml [hunk 1]
```diff
@@ -17,14 +17,13 @@ paths:
         '202':
           description: Accepted
       parameters:
-        - name: X-Sent-From
+        - name: X-Sent-By
```
Comment (joonasroosalung) [inline docs/technical/api-spec.yml:20]: Isn't 'From' a kind of standard? Why the change?

Comment (tonistuulNG) [reply docs/technical/api-spec.yml:20]: Discussed it with Rainer and:
send to / sent by sounds better than 
send to / sent from

Comment (joonasroosalung) [reply docs/technical/api-spec.yml:20]: Will it also be changed in DMR? https://github.com/buerokratt/DMR/blob/feature/dmr-implementation/src/Dmr.Api/Models/HeadersInput.cs

Comment (tonistuulNG) [reply docs/technical/api-spec.yml:20]: yes, everywhere

Comment (priyankasharma55) [reply docs/technical/api-spec.yml:20]: Possible to add the JSON schema for body for the message as well? So that it can be easily referenced.

**Review by martinkearn — APPROVED** [review]

**Review by guypritchard — APPROVED** [review]
Just one question about the reasoning behind classification field in the body for my idiot brain.

File: docs/technical/api-spec.yml [hunk 2]
```diff
@@ -33,42 +32,108 @@ paths:
...
+                  example: Greetings from Estonia!
+                classification:
+                  type: string
+                  example: NLib
+                  readOnly: true
+                context:
+                  type: object
+              required:
+                - message
+                - classification
```
[the remark this answers is not in the document]
Comment (turnerrainer) [reply docs/technical/api-spec.yml:68]: Yes. If we want to change that to original flow (or use them in parallel), the Classifier (or any other participant) would just set "X-Send-To" headers accordingly and everything would still work.

**Review by turnerrainer — APPROVED** [review]

File: docs/technical/api-spec.yml [hunk 3]
```diff
@@ -33,42 +32,108 @@ paths:
...
-      title: ClassificationRequest
-      type: array
-      items:
-        type: string
-        example: Greetings from Estonia!
-    ClassificationResponse:
-      title: ClassificationResponse
+    Classification:
+      title: Classification
+      x-stoplight:
```
[the remark this answers is not in the document]
Comment (tonistuulNG) [reply docs/technical/api-spec.yml:75]: good catch, im actually exporting with "remove stopligh tags", so if any come through it is usually because I have modified the underlying structure manyally, will fix

File: docs/technical/api-spec.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tonistuulNG) [reply docs/technical/api-spec.yml:68]: Yeah, as I understand DMR will be the glue between different Bürokratt components (some of which might not have anything to do with classifications or chats), so it feels wrong to add this info to headers.

**Comment by martinkearn** [comment]
> Just one question about the reasoning behind classification field in the body for my idiot brain.

@guypritchard I think it on the basis that DMR should not know anything about the message content or its classification for PII reasons and on the basis that the body will eventually be encrypted, having the classification there is safer than having it in plain text as a header. Dmr does not need to know about the classification, it only needs to know where to send the message on to
